chemprop/cli/hyperopt.py

```diff
@@ -1,16 +1,348 @@
 
 class HyperoptSubcommand(Subcommand):
     COMMAND = "hyperopt"
     HELP = "perform hyperparameter optimization on the given task"
 
     @classmethod
     def add_args(cls, parser: ArgumentParser) -> ArgumentParser:
-        pass
+        parser = add_common_args(parser)
+        parser = add_train_args(parser)
+        return add_hyperopt_args(parser)
 
     @classmethod
     def func(cls, args: Namespace):
-        pass
+        args = process_common_args(args)
+        args = process_train_args(args)
+        args = process_hyperopt_args(args)
+        validate_common_args(args)
+        validate_train_args(args)
+        main(args)
+
+def add_hyperopt_args(parser: ArgumentParser) -> ArgumentParser:
+
+    hyperopt_args = parser.add_argument_group("Hyperparameter optimization arguments")
+
+    hyperopt_args.add_argument(
+        "--num-samples",
+        type=int,
+        default=10,
+        help="Number of hyperparameter optimization samples to run",
+    )
+
+    hyperopt_args.add_argument(
+        "--hyperopt-seed",
+        type=int,
+        default=0,
+        help="The initial seed used for choosing parameters in hyperparameter optimization trials. In each trial, the seed will be increased by one, skipping seeds previously used.",
+    )
+
+    hyperopt_args.add_argument(
+        "--hyperopt-save-dir",
+        type=Path,
+        help="Directory to save the hyperparameter optimization results",
+    )
+
+    hyperopt_args.add_argument(
+        "--hyperopt-checkpoint",
+        type=Path,
+        help="Path to a directory where hyperopt completed trial data is stored. Hyperopt job will include these trials if restarted. Can also be used to run multiple instances in parallel if they share the same checkpoint directory."
+    )
+
+    hyperopt_args.add_argument(
+        "--startup-random-iters",
+        type=int,
+        help="The initial number of trials that will be randomly specified before TPE algorithm is used to select the rest. By default will be half the total number of trials.",
+    )
+
+    hyperopt_args.add_argument(
+        "--manual-trial-dirs",
+        type=Path,
+        nargs="+",
+        help="Paths to save directories for manually trained models in the same search space as the hyperparameter search. Results will be considered as part of the trial history of the hyperparameter search.",
+    )
+
+    hyperopt_args.add_argument(
+        "--search-parameter-keywords",
+        type=str,
+        nargs="+",
+        default=["basic"],
+        help=f"""The model parameters over which to search for an optimal hyperparameter configuration.
+    Some options are bundles of parameters or otherwise special parameter operations.
+
+    Special keywords:
+        basic - the default set of hyperparameters for search: depth, ffn_num_layers, dropout, and linked_hidden_size.
+        linked_hidden_size - search for hidden_size and ffn_hidden_size, but constrained for them to have the same value.
```

Comment:
Bringing some offline discussion here - we should remove the `linked_hidden_size` option and just make the default to optimize both.

Reply:
Removed!